enhancements/proposals/node-autoreplacement.md

```diff
@@ -0,0 +1,133 @@
+Setting `replace-address-first-boot` results in a desired output, i.e. a node which lost its storage will be replaced during bootstrap.
+Otherwise, setting the flag is safe as Scylla ignores it after bootstrap.
+
+Introducing this enhancement additionally allows for removing replacement functionality from `OrphanedPVController`. The controller would be simplified and only be responsible for removing the orphaned PVs.
+
+### User Stories
+
+#### Story 1
+
+The user is running a ScyllaCluster and looses a PVC bound to one of Scylla nodes. The node gets automatically replaced. 
+
+#### Story 2
+
+The user is running a ScyllaCluster on AWS/GCP cluster with a local storage. One of the instances is subject to an event resulting in erasing the local storage (see [AWS Instance store lifetime](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-lifetime) and [GCP Local SSD data persistance](https://cloud.google.com/compute/docs/disks/local-ssd#data_persistence))  .
+A node scheduled on this instance loses its storage, even though a PVC bound to it is not removed. The node gets replaced automatically.
+
+### Notes/Constraints/Caveats [Optional]
+
+- It would be preferred to use [Immutable ConfigMaps](https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-immutable).
+- Having introduced this feature, we might consider removing the existing functionality of [replacing dead nodes manually](https://operator.docs.scylladb.com/master/nodeoperations/replace_node.html).
+
+### Risks and Mitigations
+
+- The proposed design is based on a fact that `replace-address-first-boot` flag is ignored after bootstrap. Any changes to this behaviour or side effects introduced could result in an undefined behaviour of the enhancement.
+
+## Design Details
+
+To check whether the node is not the first to bootstrap in a cluster and already exists in gossip, the controller uses [`ScyllaClient's Status`](https://github.com/scylladb/scylla-operator/blob/689c0a0a82aeea0fef72b2cd11af0333d1d2e180/pkg/scyllaclient/client.go#L98).
+Client is created with default config. Controller authenticates with the already known token. It communicates with all nodes of a given cluster. The conditions for creating the `ConfigMap` are met if the node's IP is among the returned hosts.
+
+`Sidecar` accesses the `ConfigMap` through a dedicated `Volume`.
+
+### Test Plan
+
+- e2e/integration tests
+     1.
+          - Create a `ScyllaCluster` with 2 nodes
+          - Wait for the `ScyllaCluster` to deploy
+          - Remove first node's PVC
+          - Remove first node's Pod
+          - Await PVC's and Pod's deletion
+          - Wait for the `ScyllaCluster` to redeploy
+          - Verify `ScyllaCluster`
+     2.
+          - Create a `ScyllaCluster` with 2 nodes
+          - Wait for `ScyllaCluster` to deploy
+          - Turn on maintenance mode, Exec `systemctl stop scylla && rm -rf /var/lib/scylla` in first node's Pod
```

Comment:
is this simulating hot swapping disks? would a scylla process be stopped in that case? otherwise this would look pretty similar to the first case where a PVC is deleted and it get new and empty disk.

Reply:
You're right. I've removed the second test.